### PR TITLE
Add Model Analyzer to Triton SDK Container

### PR DESCRIPTION
**Pull request by Tabrizian**
It should be merged after 20.11 code freeze.

### CODE REVIEW

File: Dockerfile.sdk
Comment (dzier) [inline Dockerfile.sdk:32]: This will need to be `main` branch before we merge this change

File: Dockerfile.sdk
Comment (dzier) [inline Dockerfile.sdk:40]: We may have to hold off on this for now. If we go with NVML, then all of this new installation is not needed.

File: Dockerfile.sdk
Comment (dzier) [inline Dockerfile.sdk:121]: If there is a top level setup.ini file, you don't need to clone the repo. You can just use:
```
ARG TRITON_MODEL_ANALYZER_REPO=https://github.com/triton-inference-server/model_analyzer@${TRITON_MODEL_ANALYZER_REPO_TAG}

pip install "git+${TRITON_MODEL_ANALYZER_REPO}"
```

This is one way we will instruct people install the model analyzer.

File: Dockerfile.sdk
Comment (dzier) [inline Dockerfile.sdk:128]: With pip install, this is not needed

**Review by dzier — CHANGES_REQUESTED** [review]
Note that this will be used for how we release Model Analyzer.

Also, this does not rename the Triton client containers, so the PR title should change. You should separate adding Model Analyzer from renaming the container in two different PRs / commits.

File: Dockerfile.sdk
Comment (Tabrizian) [reply Dockerfile.sdk:128]: This was in the Dockerfile before. I think this is because of the `perf_analyzer`.

File: Dockerfile.sdk
Comment (CoderHam) [reply Dockerfile.sdk]: Please add TRITON_MODEL_ANALYZER_REPO_TAG to the gitlab yml (gitlab PR)

Comment (Tabrizian) [reply Dockerfile.sdk]: It is in the gitlab PR.

File: Dockerfile.client
Comment (deadeyegoodwin) [inline Dockerfile.client:40]: Why not an ARG? Is this envvar needed when the container is running?

File: Dockerfile.client
Comment (deadeyegoodwin) [inline Dockerfile.client:121]: Why not an ARG? Is this envvar needed when the container is running?

**Review by deadeyegoodwin — CHANGES_REQUESTED** [review]

**Review by deadeyegoodwin — APPROVED** [review]

**Review by deadeyegoodwin — APPROVED** [review]

**Review by deadeyegoodwin — APPROVED** [review]

**Pull request merged**
